File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.js
function solution(code) {
    let answer = '';
    let mode = 0;
    for (let i = 0; i < code.length; i++){
        if(code[i] ==='1') {
            // switch mode between 1 and 0;
            mode = 1 - mode;
            // check i is even or odd using mode; 
            // mode 0 returns i is even, mode 1 returns i is odd
        } else if (i % 2 === mode) {
            answer += code[i];
        }
    }
    // Use OR logical operator because '' is false
    // Shorter than return answer === '' ? 'EMPTY' : answer
    return answer || 'EMPTY';
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    mode = 0
    ret = ''
    for i, c in enumerate(code):
        if c == '1':
            mode = 1 - mode
        else:
            if mode == 0 and i % 2 == 0:
                ret += c
            elif mode == 1 and i % 2 == 1:
                ret += c
    return ret if ret else 'EMPTY' 